### ISSUE TITLE
Cannot connect to Nissan Connect (EU) service. Nissan Ariya.

### ISSUE THREAD

**Issue by SpainBoys** [dan-r/HomeAssistant-NissanConnect#19]
I own a Nissan Ariya from Spain. 
I have installed your module. When I try to connect to the Nissan Connect (EU) service, I get an error (see screenshot)
I enter login and password the same as in NissanConnect Services application.
I tried to find this error in the HomeAssistans logs, but I could not find it anywhere.
I would like your help in solving this problem.
Thank you.

![image](https://github.com/dan-r/HomeAssistant-NissanConnect/assets/157040441/5bbff624-390c-432e-82a8-199480cc368a)

**Comment by dan-r** [comment]
How are you running Home Assistant? Docker?

**Comment by SpainBoys** [comment]
> How are you running Home Assistant? Docker?

No, I do not use Docker . 
I have HomeAssistant installed in a virtual machine (haos_ova) on a laptop with an Intel processor.
All the latest HomeAssistant updates are installed. 
Your NissanConnect for Home Assistant integration is version 0.5.0

**Comment by SpainBoys** [comment]
> How are you running Home Assistant? Docker?

You may find this information useful in understanding why I am not logging in through your integration.
When logging into the NissanConnect Services application, you are prompted to connect to this resource

https://prod.eu2.auth.kamereon.org

Then I enter my login and password and my Nissan Ariya car is connected in the application and I can control it and see statistics about trips and charging.

**Comment by dan-r** [comment]
Thats the same as I see and it works for me... Shouldn't make a difference but do you have any non-ASCII characters in your password?

**Comment by SpainBoys** [comment]
> Thats the same as I see and it works for me... Shouldn't make a difference but do you have any non-ASCII characters in your password?

Yes, my password has only one @ symbol in it. 
All other numbers and letters in the password.
Are you suggesting I change my password?

**Comment by dan-r** [comment]
You can view the logs from Home Assistant settings (Settings > System > Logs) or from a log file in your virtual machine - it would depend how you've set it up as to where it is.

**Comment by mxg7652** [comment]
Hey, i tried setting up the integration today and faced the same issue. 
I have a 2020 Leaf in Germany and use the NissanConnect Services app as well, so the credentials should be correct. 
I run HA in Docker and couldn't find any error in the logs related to this integration.

**Comment by 0cutaway** [comment]
Hi @dan-r ,

Great integration! Was worried this wouldn't exist, so thanks for making it happen!

Installed it a few days ago and connected fine with the Ariya. My only observation I can tell is that the temperature shows it updating at 5pm everyday. there are no other data points. I left the polling intervals as default. Is this normal?

I haven't driven the car this week so can't tell if its the same for the battery % data

Thanks

**Comment by dan-r** [comment]
@SpainBoys and @mxg7652 would you mind trying now? There were some breaking API changes this week (#23), so maybe it works now...?

**Comment by dan-r** [comment]
@0cutaway thank you! There is a bit of weirdness with the Ariya data. It uses some different API endpoints for some of this data. It all needs a bit more investigation

**Comment by SpainBoys** [comment]
> @SpainBoys and @mxg7652 would you mind trying now? There were some breaking API changes this week (https://github.com/dan-r/HomeAssistant-NissanConnect/issues/23), so maybe it works now...?

Updated your integration to version 0.5.1
Rebooted HA and tried to enter my NissanConnect login and password, but the error is still the same as my original request - "Invalid credentials provided.".
There is no error in the logs as before.
It is very strange to me why some owners can connect through your interface and some can't....
Can you configure your integration to write logs while connecting to NissanConnect system, so that you can understand at least what the error is....

**Comment by dan-r** [comment]
v0.5.2-beta.1 adds logging to the login function

**Comment by SpainBoys** [comment]
> v0.5.2-beta.1 adds logging to the login function

Thank you for your promptness!
Updated your integration and tried to log in again.
This is the error that appears now.
![image](https://github.com/dan-r/HomeAssistant-NissanConnect/assets/157040441/3ee014b6-4a27-44df-a88f-69a0f64bc3df)
**2024-05-17 16:34:55.683 ERROR (SyncWorker_8) [custom_components.nissan_connect.kamereon] Invalid credentials provided: {"code":401,"reason":"Unauthorized","message":"Authentication Failed"}**

**Comment by mxg7652** [comment]
Hey, thanks for looking into it. I am getting the same error as SpainBoys:

`2024-05-17T15:41:35.539597897Z 2024-05-17 17:41:35.538 ERROR (SyncWorker_19) [custom_components.nissan_connect.kamereon] Invalid credentials provided: {"code":401,"reason":"Unauthorized","message":"Authentication Failed"}`

**Comment by SpainBoys** [comment]
Good morning!
That's it, I managed to connect in your integration.
Problem solved.

I entered my password incorrectly. It turned out that the password substitution in the phone put the correct password in the application, and I apparently went to the wrong place and took a similar password, so it did not let me log in through your integration.
I apologize for my mistake. 

Thank you for such a great integration of NissanConnect into HomeAssistant
This topic can be closed.
